workflow_creation/utilities_auto_gui.md

```diff
@@ -1,96 +1,81 @@
+The _app_ directory contains a configuration file for the web site layout parameters such as font size or title size. The listing below shows a complete layout configuration file. 
+
+```XML
+<pageConfiguration>
+    <params>
+        <param name="layout">top2bottom</param> <!-- Specify the order of GUI elements (top2bottom or left2right)-->
+        <param name="orientation">single</param>
+		<param name="margin">20</param>
+		<param name="margin_direction">left</param>
+		<param name="title_size">4</param> <!-- from 1 (large) to 6 (small) -->
+		<param name="heading_size">5</param> <!-- from 1 (large) to 6 (small) -->
+		<param name="font_size">14px</param> <!-- Size for text other than title and headings -->
+    </params>
+</pageConfiguration>
+```
+
+#### 2.3 Excel-Sheet for GUI Elements:
+The _app_ directory also contains an excel sheet, which lists all gui elements. It allows to specify *element name*, *element type*, *unit of measurement*, *default value*, *min/max values*, *checked* state and the name of the *output parameter* for the workflow manager. The min/max parameters can be used to prevent users from specifying malicious parameters but are not mandatory. The checked state option is only used for checkable elements, i.e., checkboxes and radiobuttons. The unit of measurement option can be used to present the unit, e.g., milimeters, of the parameter to the user. Note that the unit of measurement option is mandatory for radiobuttons, since the application assigns radiobuttons of the same unit to the same group. For an complete example of a valid excel stylesheet checkout the *GUIAppParameters.xlsx* file in the *app* directory.
+
+Currently the application supports the following data types:
++ **title**: The title of the web page. This shall be set first.
++ **heading**: Any heading to inform users about the semantics of GUI elements
++ **integer**: Integer input field
++ **decimal**: Float input field
++ **string**: String input field
++ **3 decimals**: Three contiguous decimal input fields
++ **6 decimals**: Six contiguous decimal input fields
++ **radiobutton**: A single radiobutton. To form groups of radiobuttons specify same unit of measurement. Set the value of the *checked* field to *YES* for an initially checked radiobutton.
++ **checkbox**: A single checkbox. Set the value of the *checked* field to *YES* for an initially checked checkbox.
+
+
+3 Deployment
+---
+Before service deployment ensure that docker is installed and the service configuration is correct. The deployment of the application just amounts to building and starting the docker container. From the directory containing this README file execute the following:
+
+```
+docker build -t <container_name> .
+docker run -d -p <host_port>:80 --env-file=env --name <container_name> <container_name>
+```
+
+The application runs automatically on the docker container.
+
+Finally, connect the outputs of the GUI application with the input ports of the appropriate services.

```

Comment:
Before this can be done, the service needs to be registered on the platform.

Reply:
Good point! I also added sentence on how the wsdl location of the service is structured.